Tests/UnitTests/AdyenTheme/AdyenThemeTextFieldTests.swift

```diff
@@ -0,0 +1,198 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) @testable import AdyenUI
+import XCTest
+
+final class AdyenThemeTextFieldTests: XCTestCase {
+    
+    func test_textFieldMethod_defaultTheme() {
+        // Given
+        let defaultTheme = AdyenTheme()
+        
+        // Then
+        XCTAssertEqual(defaultTheme.textFieldStyle.backgroundColor, AdyenColorScheme.default.background)
+        XCTAssertEqual(defaultTheme.textFieldStyle.textColor, AdyenColorScheme.default.text)
+        XCTAssertEqual(defaultTheme.textFieldStyle.activeColor, AdyenColorScheme.default.highlight)
+        XCTAssertEqual(defaultTheme.textFieldStyle.errorColor, AdyenColorScheme.default.destructive)
+        XCTAssertEqual(defaultTheme.textFieldStyle.cornerRadius, CornerRounding.fixed(AdyenUIConstants.defaultCornerRadius))
+        XCTAssertEqual(defaultTheme.textFieldStyle.borderColor, AdyenColorScheme.default.outline)
+        XCTAssertEqual(defaultTheme.textFieldStyle.borderWidth, AdyenUIConstants.defaultBorderWidth)
+    }
+
+    func test_textfieldMethod_shouldUpdateBackgroundColor() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        newTextFieldStyle.backgroundColor = AdyenColorScheme.default.primary
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.backgroundColor, AdyenColorScheme.default.primary)
+    }
+    
+    func test_textfieldMethod_shouldUpdateTextColor() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        newTextFieldStyle.textColor = AdyenColorScheme.default.textOnPrimary
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.textColor, AdyenColorScheme.default.textOnPrimary)
+    }
+    
+    func test_textfieldMethod_shouldUpdateActiveColor() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        newTextFieldStyle.activeColor = AdyenColorScheme.default.highlight
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.activeColor, AdyenColorScheme.default.highlight)
```

Comment:
`AdyenColorScheme.default.highlight` is a default color for `activeColor`

Reply:
As one of the ways to exclude occasional matching of default colors and to ensure test is useful, I
d suggest using simple `UIColor` predefined colors that rarely used in designs like `.pink` or `.purple`.
And as my previous comment saids, let's use `let expectedValue = UIColor.pink` to make reading and copy-pasting easier.